packages/react/src/lib/createStardustComponent.tsx

```diff
@@ -18,13 +20,13 @@ export interface CreateStardustComponentConfig<P> {
   actionHandlers?: AccessibilityActionHandlers
 }
 
-const createComponent = <P extends {} = {}, S extends {} = {}>({
+const createComponent = <P extends {} = ObjectOf<any>>({
```

Comment:
I was also considering the more correct fix:
```suggestion
const createComponent = <P extends {} = React.HTMLAttributes<HTMLElement>>({
```
but for some reason we don't seem to be fans of typing our components as extensions of HTML elements 😕

Reply:
yes - lets just start with loose but correct types for now, and steadily restrict them after. The same suggestion here:
```js
const createComponent = <P extends ObjectOf<any> = {}>({ ...
```

Reply:
see https://github.com/stardust-ui/react/pull/954#discussion_r260265890

Reply:
yes - as a result, have agreed with @Bugaa92 to introduce the following type:

```js
const createComponent = <P extends ObjectOf<any> = {}>({ ..
```
This will prevent the following things:
```js
const Foo = createComponent<number>(...)
const Foo = createComponent<string>(...)
```
and, at the same time, will work properly for the scenarios @Bugaa92 has mentioned 👍

Reply:
after discussions with @kuzhelov we agreed to go with `P extends ObjectOf<any> = any`